### PR TITLE
Create application

### PR DESCRIPTION
**Pull request by bonustrack**
Here is how it look to create a proxy account for a new app:
create-app

### CODE REVIEW

File: src/utils/validator.js [hunk 1]
```diff
@@ -16,3 +16,53 @@ export const accountExist = (rule, value, callback) => {
...
+    if (result[0]) {
+      callback(['Account name is not available']);
+    } else {
+      callback();
+    }
+  });
+};
+
+// https://github.com/steemit/condenser/blob/eaf8a02658b8deaef376ec90b81d0866e52582cc/app/utils/ChainValidation.js#L4
+export const validateAccountName = (rule, value, callback) => {
```
Comment (nil1511) [inline src/utils/validator.js]: Would be better to move this to steem.js.

File: src/components/Apps/CreateApp.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+          /** Send request to server for create app */
+          fetch(`/api/apps/@${clientId}`, {
+            headers: new Headers({
+              Authorization: this.props.auth.token,
+            }),
+            method: 'POST',
+          })
+            .then(res => res.json())
+            .then((data) => {
+              if (!data.error) {
```
Comment (nil1511) [inline src/components/Apps/CreateApp.js]: Is error sent with status code 200? If not they should be handled in the catch.

File: src/components/Apps/CreateApp.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    const accountCreationFee = await getAccountCreationFee();
+
+    /** Generate account authorities */
+    const publicKeys = config.offline_generated_public_keys;
+    const owner = { weight_threshold: 1, account_auths: [['steemconnect', 1]], key_auths: [[publicKeys.owner, 1]] };
+    const active = { weight_threshold: 1, account_auths: [['steemconnect', 1]], key_auths: [[publicKeys.active, 1]] };
+    const posting = { weight_threshold: 1, account_auths: [['steemconnect', 1]], key_auths: [[publicKeys.posting, 1]] };
+
+    /** Create proxy account */
+    steem.broadcast.accountCreateWithDelegation(
```
Comment (nil1511) [inline src/components/Apps/CreateApp.js]: Using promise version(`await`) is more preferred in that way `handleSignFormSubmit` can be thenable if needed.

File: src/components/Apps/CreateApp.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+  showModal = () => {
+    this.setState({ visible: true });
+  };
+
+  hideModal = () => {
+    this.setState({ visible: false });
+  };
+
+  handleFormSubmit = async (values) => {
```
Comment (nil1511) [inline src/components/Apps/CreateApp.js]: `async` can be removed here

File: routes/apps.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  }
+  res.json(app);
+});
+
+/** Create application */
+router.post('/@:clientId', authenticate('user'), async (req, res, next) => {
+  const { clientId } = req.params;
+
+  /** Wait 5 seconds to insure the newly created account is available on the node */
+  await sleep(5000);
```
Comment (nil1511) [inline routes/apps.js]: Should not delay API. Would be better to delay on front-end `CreateApp.js:75`
It should throw an error if the account is not available with msg indicating account creation can take upto 5 sec. That way developer would add sleep in front-end if he missed it.

File: src/utils/validator.js (hunk 1 above)
Comment (bonustrack) [inline src/utils/validator.js]: I will change later when we will have the method shipped in new steem version

**Pull request merged**